src/API/Middlewares/HttpExceptionsMiddleware.ts

```diff
@@ -14,10 +14,13 @@ export const HttpExceptionsMiddleware: Koa.Middleware = async (context: any, nex
       context.status = 404
     } else {
       if (context.logger)
-        context.logger.error({ exception }, 'Error Caught at Middleware - Will Return Internal Server Error')
+        context.logger.error({ exception }, 'Error Caught at Middleware')
 
-      context.body = 'Internal Server Error'
-      context.status = 503
+      context.body = {
+        message: exception.message || 'Internal Server Error',
```

Comment:
Instead of just returning a string for the body, I think it's better we return json with a message property, that allows us to all return additional details in the future as well.

Reply:
It takes us away from using `response.text()` but I do prefer structured errors returned.